### DISCUSSION TITLE
When is Analysis data recomputed

### DISCUSSION THREAD

**Discussion by meithecatte** [egraphs-good/egg#238] [Q&A]
`Analysis::make` is passed a reference to the entire e-graph, so in theory it could depend on anything at all in the e-graph. Of course, then, there is no hope of keeping the analysis data up-to-date. I've noticed that changes in the analysis data of an e-class cause the analysis to be recomputed for all the parent nodes, and so on, recursively upwards. Is it kosher, though, for an analysis to inspect the structure of the children nodes directly? Will unions affecting those classes cause a ripple of recomputation, or will it that depend on the hints returned by `DidMerge`?

**Comment by mwillsey** [comment]
It will depend entirely on the DidMerge. Your intuition is correct; while you _can_ look at "non-local" information, only the DidMerge tells the systems when to stop propagating.
